### DISCUSSION TITLE
Removing duplicates after migration

### DISCUSSION THREAD

**Discussion by maratsh** [immich-app/immich#4389] [Q&A]
I've just started using immich. It is amazing product, thank you.

I did some time testing application for myself and installed mobile application with sync to server. 
At the same time I had another solution, also syncing data to it's own server.
Ive decided to migrate to immich from another solution. I added another folder library path to my user and hit 'scan'. After some hours, I see all images appear in my library, awesome.

 However, there is a problem. As I was running both solutions on the same devices/devices, *some* assets are duplicated now.

How I can deduplicate them now? any suggestion? I come up with this SQL to get ID's

```
SELECT
  *
FROM
  assets
WHERE
  assets.type='IMAGE' AND
  assets."ownerId"='xxx-xxx-xxx-xx' AND
  assets."originalPath" like '%upload/library/path/%' AND
  assets."originalFileName" IN (
    Select
      a."originalFileName"
    from
      (
        SELECT
          assets."originalFileName",
          count(1) over(
            PARTITION by assets."originalFileName",
            assets.checksum
          ) as Cnt
        FROM
          assets
      ) a
    Where
      Cnt > 1
  );

```

How I can delete them now? should I just delete them from assets table?

**Comment by agross** [comment]
You can try what I wrote to get rid of my duplicates: https://github.com/agross/immich-duplicates/
